logic/transactionPool.js

```diff
@@ -529,6 +529,20 @@ TransactionPool.prototype.queueTransaction = function (transaction, cb) {
 	return setImmediate(cb);
 };
 
+/**
+ * Adds transaction to multisignature or queue list.
+ * @implements {addMultisignatureTransaction}
+ * @implements {addQueuedTransaction}
+ * @param {transaction} transaction
+ */
+TransactionPool.prototype.addReady = function (transaction) {
+	if (transaction.type === transactionTypes.MULTI || Array.isArray(transaction.signatures)) {
+		self.addMultisignatureTransaction(transaction);
+	} else {
+		self.addQueuedTransaction(transaction);
+	}
+};
```

Comment:
Why not use already defined: `TransactionPool.prototype.queueTransaction`?

Reply:
Because `queueTransaction` overrides the original `receivedAt` value and also potentially can reject the transaction because it checks pool availability.
Do you think we should use the same path for regular transactions in this scenario?

Reply:
Yes, I think the transaction should follow the same path from where it came from.

Example, maybe the pool is full, therefore we should not bypass the check.

Reply:
Ok, I will use the regular path to put transactions back into transaction pool after block deletion